internal/catalog/internal/controllers/failover/controller_test.go

```diff
@@ -4,336 +4,302 @@
 package failover
 
 import (
-	"context"
 	"fmt"
 	"testing"
 
-	"github.com/stretchr/testify/suite"
-
-	svctest "github.com/hashicorp/consul/agent/grpc-external/services/resource/testing"
-	"github.com/hashicorp/consul/internal/catalog/internal/mappers/failovermapper"
 	"github.com/hashicorp/consul/internal/catalog/internal/types"
 	"github.com/hashicorp/consul/internal/controller"
+	"github.com/hashicorp/consul/internal/controller/controllertest"
 	"github.com/hashicorp/consul/internal/resource"
+	"github.com/hashicorp/consul/internal/resource/resourcetest"
 	rtest "github.com/hashicorp/consul/internal/resource/resourcetest"
 	pbcatalog "github.com/hashicorp/consul/proto-public/pbcatalog/v2beta1"
 	"github.com/hashicorp/consul/proto-public/pbresource"
-	"github.com/hashicorp/consul/sdk/testutil"
 )
 
-type controllerSuite struct {
-	suite.Suite
+func TestController(t *testing.T) {
```

Comment:
It may not look that way from this diff but this test was mostly untouched except for the setup and moving this out of a test suite and into just a single test. The test previously wasn't asserting anything about the state of the failover mapper so there wasn't much to modify here other than to ensure the test configures the cache for the controller.